analyze/analyses/garbage.rs

```diff
@@ -23,9 +24,19 @@ impl traits::Emit for Garbage {
             (Align::Right, "Size %".to_string()),
             (Align::Left, "Garbage Item".to_string()),
         ]);
+        let items_iter = self.items.iter().map(|id| &items[*id]);
+        let mut hidden_count = 0;
+        let mut hidden_size = 0;
+
+        for item in items_iter.clone().take(self.limit) {
+            // Skip this item if it is a data segment and we're summarizing them later
+            if !self.show_data_segments && item.name().starts_with("data") {
```

Comment:
There is a better way to determine if this item is a data segment or not:

* add a getter for an item's kind (we just haven't needed one yet): `pub fn kind(&self) -> &ItemKind` to `impl Item { .. }` in `ir/ir.rs`

* and then here, do something like:

```rust
if let ItemKind::Data(_) = item.kind() {
    // ...
}
```

Reply:
Wow, I can't believe I missed that! Thanks for pointing it out :)